ts/screens/messages/paginated/MessageDetailScreen.tsx

```diff
@@ -91,8 +91,13 @@ const MessageDetailScreen = ({
     if (!isRead) {
       setMessageReadState(message.id, true);
     }
-    loadMessageDetails(message.id);
-  }, []);
+    if (
+      pot.isError(messageDetails) ||
+      (pot.isNone(messageDetails) && !pot.isLoading(messageDetails))
+    ) {
+      loadMessageDetails(message.id);
+    }
+  }, [isRead, message.id, messageDetails]);
```

Comment:
With the current implementation, in case of error, we have an infinite loop:



https://user-images.githubusercontent.com/26501317/144033727-198b4b28-c044-40f6-ab72-7dfc9dd21554.mov


In our case we should bootstrap the `loadMessageDetails` only at the first rendering if the pot is not `some` and after that give the control to the user if we still have an error, this is my try:
```typescript
  const firstRender = React.useRef<boolean>(true);
  useEffect(() => {
    if (!isRead) {
      setMessageReadState(message.id, true);
    }
    const shouldRequest = firstRender.current && !pot.isSome(messageDetails);
    if (shouldRequest) {
      // eslint-disable-next-line functional/immutable-data
      firstRender.current = false;
      loadMessageDetails(message.id);
    }
  }, [isRead, loadMessageDetails, message.id, messageDetails, setMessageReadState]);
```

Also is worth to add `loadMessageDetails` and `setMessageReadState` as hook dependencies, in order to avoid the ESLint warning.